bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptEngineManager.java

```diff
@@ -16,47 +16,48 @@
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.module.script.internal.provider.ScriptModuleTypeProvider;
 
 /**
+ * The ScriptEngineManager provides the ability to load and unload scripts.
  *
  * @author Simon Merschjohann - Initial contribution
+ * @author Scott Rushworth - changed parameter names when implementing {@link ScriptModuleTypeProvider}
  */
 @NonNullByDefault
 public interface ScriptEngineManager {
 
     /**
-     * Checks if a given fileExtension is supported
+     * Creates a new ScriptEngine used to execute scripts, ScriptActions or ScriptConditions
      *
-     * @param fileExtension
-     * @return true if supported
+     * @param scriptType       a file extension (script) or MimeType (ScriptAction or ScriptCondition)
+     * @param engineIdentifier the unique identifier for the ScriptEngine (script file path or UUID)
+     * @return ScriptEngineContainer or null
      */
-    boolean isSupported(String fileExtension);
+    @Nullable
+    ScriptEngineContainer createScriptEngine(String scriptType, String engineIdentifier);
 
     /**
-     * Creates a new ScriptEngine based on the given fileExtension
+     * Loads a script and initializes its scope variables
      *
-     * @param fileExtension
-     * @param scriptIdentifier
-     * @return
+     * @param engineIdentifier the unique identifier for the ScriptEngine (script file path or UUID)
+     * @param scriptData       the content of the script
      */
-    @Nullable
-    ScriptEngineContainer createScriptEngine(String fileExtension, String scriptIdentifier);
+    void loadScript(String engineIdentifier, InputStreamReader scriptData);
 
     /**
-     * Loads a script and initializes its scope variables
+     * Unloads the ScriptEngine loaded with the engineIdentifier
      *
-     * @param fileExtension
-     * @param scriptIdentifier
-     * @param scriptData
-     * @return
+     * @param engineIdentifier the unique identifier for the ScriptEngine (script file path or UUID)
      */
-    void loadScript(String scriptIdentifier, InputStreamReader scriptData);
+    void removeEngine(String engineIdentifier);
 
     /**
-     * Unloads the ScriptEngine loaded with the scriptIdentifer
+     * Checks if the supplied file extension or MimeType is supported by the existing ScriptEngineFactories
      *
-     * @param scriptIdentifier
+     * @param scriptType a file extension (script) or MimeType (ScriptAction or ScriptCondition)
+     * @return boolean
```

Comment:
IIRC this has been already comment. The JavaDoc should contain a description of the return value, not the type.
Why not using `true if supported` is in the old `isSupported` method?

Reply:
I fixed the other but forgot this one. I'll fix this...

Reply:
Done